### PR TITLE
Elasticsearch: Fix using of individual query time ranges when querying

### PR DESCRIPTION
**Pull request by ivanahuckova**
In this PR, we are removing creating of indices in client and instead, we create them when creating individual search elasticsearch queries, so we can use individual query time range (which is for example available in alerting).

Fixes https://github.com/grafana/support-escalations/issues/9553

### CODE REVIEW

File: pkg/tsdb/elasticsearch/client/client.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type Client interface {
 }
 
 // NewClient creates a new elasticsearch client
-var NewClient = func(ctx context.Context, ds *DatasourceInfo, timeRange backend.TimeRange, logger log.Logger, tracer tracing.Tracer) (Client, error) {
+var NewClient = func(ctx context.Context, ds *DatasourceInfo, queries []backend.DataQuery, logger log.Logger, tracer tracing.Tracer) (Client, error) {
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply pkg/tsdb/elasticsearch/client/client.go:59]: Yes, you are right.

**Comment by ivanahuckova** [comment]
Okay, I will polish this solution and add + update tests.

**Review by svennergr — APPROVED** [review]
I think we should add that one log line back - at least similar to what we had. Rest LGTM!

File: pkg/tsdb/elasticsearch/client/client.go [hunk 2]
```diff
@@ -65,19 +64,12 @@ var NewClient = func(ctx context.Context, ds *DatasourceInfo, timeRange backend.
 		return nil, err
 	}
 
-	indices, err := ip.GetIndices(timeRange)
-	if err != nil {
-		return nil, err
-	}
-	logger.Debug("Creating new client", "configuredFields", fmt.Sprintf("%#v", ds.ConfiguredFields), "indices", strings.Join(indices, ", "), "interval", ds.Interval, "index", ds.Database)
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply pkg/tsdb/elasticsearch/client/client.go:72]: Yeah, accidental. Bringing it back. Great catch.

[the remark this answers is not in the document]
Comment (ivanahuckova) [reply pkg/tsdb/elasticsearch/client/client.go:72]: Fixed in https://github.com/grafana/grafana/pull/84201/commits/a0733b6c1220cd34a2f098be1cba99b6e50f9470

**Pull request merged**